addons/new-osio-addon/src/main/java/io/fabric8/launcher/osio/steps/GitSteps.java

```diff
@@ -1,25 +1,103 @@
+    private Logger log = Logger.getLogger(GitSteps.class.getName());
 
     @Inject
     private GitService gitService;
 
+    @Inject
+    private OpenshiftClient openshiftClient;
+
+    @Inject
+    private Event<StatusMessageEvent> statusEvent;
 
     public GitRepository createRepository(OsioProjectile projectile) {
-        return null;
+        GitRepository gitRepository;
+        final String repositoryName = Objects.toString(projectile.getGitRepositoryName(), projectile.getOpenShiftProjectName());
+        if (projectile.getStartOfStep() > StatusEventType.GITHUB_CREATE.ordinal()) {
+            // Do not create, just return the repository
+            gitRepository = gitService.getRepository(repositoryName)
+                    .orElseThrow(() -> new IllegalArgumentException("Repository not found " + repositoryName));
+        } else {
+            final String repositoryDescription = projectile.getGitRepositoryDescription();
+            gitRepository = gitService.createRepository(repositoryName, repositoryDescription);
+        }
+        statusEvent.fire(new StatusMessageEvent(projectile.getId(), GITHUB_CREATE,
+                                                singletonMap("location", gitRepository.getHomepage())));
+        return gitRepository;
     }
 
 
+    public void createWebHooks(OsioProjectile projectile, GitRepository repository) {
+        if (projectile.getStartOfStep() <= StatusEventType.GITHUB_WEBHOOK.ordinal()) {
+            String jenkinsUrl = openshiftClient.getJenkinsUrl();
+            try {
+                URL webhookUrl = new URL(URLUtils.pathJoin(jenkinsUrl, "/github-webhook/"));
+                gitService.createHook(repository, null, webhookUrl, gitService.getSuggestedNewHookEvents());
+            } catch (final DuplicateHookException dpe) {
+                // Swallow, it's OK, we've already forked this repo
+                log.log(Level.FINE, dpe.getMessage(), dpe);
+            } catch (MalformedURLException e) {
+                throw new IllegalArgumentException("openshift configured jenkins url is invalid", e);
+            }
+        }
+        statusEvent.fire(new StatusMessageEvent(projectile.getId(), GITHUB_WEBHOOK));
+    }
+
+    public void pushToGitRepository(OsioProjectile projectile, GitRepository repository) {
+        if (projectile.getStartOfStep() <= StatusEventType.GITHUB_PUSHED.ordinal()) {
+            Path projectLocation = projectile.getProjectLocation();
+
+            // Add logged user in README.adoc
+            Path readmeAdocPath = projectLocation.resolve("README.adoc");
+            if (Files.exists(readmeAdocPath)) {
+                try {
+                    String content = new String(Files.readAllBytes(readmeAdocPath));
+                    Map<String, String> values = new HashMap<>();
+                    values.put("loggedUser", gitService.getLoggedUser().getLogin());
+                    String newContent = new StrSubstitutor(values).replace(content);
```

Comment:
See the osio-missioncontrol branch. the lang3 dep is not needed

Reply:
fixed in #84